### DISCUSSION TITLE
IllegalStateException: BUFR file has incomplete tables

### DISCUSSION THREAD

**Discussion by omanthey** [Unidata/netcdf-java#1395] [Questions and Answers]
Hello,

currently I'm working on importing BUFR-Files that I get from the German Weather Service (DWD). This data is originated from different federal states in Germany. As some of the files are being imported without problems I get the mentioned Exception while trying to process other files. The files are using different master table versions. Files with version 28 are fine while files with version 14 are throwing the exception. I attach a *.zip file containing two files one working and one not.

I hope that somebody can help me here as in the moment I'm totally stuck and do not know how to solve the problem.

[bufr-files.zip](https://github.com/user-attachments/files/17813026/bufr-files.zip)

p.s. Even ToolsUI (V 5.0) cannot open the file and gives the same exception

**Comment by lesserwhirls** [comment]
The issue is that `swis2-ISXD70_DWHV_LATEST-NI---bin` contains 11 element descriptors that use local tables that we do not have. It's possible we could support decoding these, but we would need the local tables.

**Comment by zfy68** [comment]
> Hello,
> 
> currently I'm working on importing BUFR-Files that I get from the German Weather Service (DWD). This data is originated from different federal states in Germany. As some of the files are being imported without problems I get the mentioned Exception while trying to process other files. The files are using different master table versions. Files with version 28 are fine while files with version 14 are throwing the exception. I attach a *.zip file containing two files one working and one not.
> 
> I hope that somebody can help me here as in the moment I'm totally stuck and do not know how to solve the problem.
> 
> [bufr-files.zip](https://github.com/user-attachments/files/17813026/bufr-files.zip)
> 
> p.s. Even ToolsUI (V 5.0) cannot open the file and gives the same exception

The local descriptor specified by DWD V=9 (GC=78/GSC=0) can be found on the official website. If it is added to the local code table, it can support parsing.
[www.dwd.de](https://www.dwd.de/EN/ourservices/bufr_extensions_national/bufr_extensions_national.html?lsId=485034&cl2Categories_LeistungsId=bufr_erweiterungen_national&cl2Categories_Auswahl=bufr_erweiterungen_national_elements)
